mesonbuild/backend/backends.py

```diff
@@ -834,7 +834,7 @@ def canonicalize_filename(fname: str) -> str:
             fname = fname.replace(ch, '_')
         return hashed + fname
 
-    def object_filename_from_source(self, target: build.BuildTarget, source: 'FileOrString') -> str:
+    def object_filename_from_source(self, target: build.BuildTarget, source: 'FileOrString', targetdir: str = None) -> str:
```

Comment:
This violates strict optional, it should be `targetdir: T.Optional[str] = None`)